src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -1,23 +1,33 @@
 using GraphQL.Server;
 using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.Extensions.DependencyInjection;
 using VirtoCommerce.ExperienceApiModule.Core.Schema;
+using VirtoCommerce.ExperienceApiModule.XOrder.Authorization;
 using VirtoCommerce.ExperienceApiModule.XOrder.Schemas;
+using VirtoCommerce.Platform.Core.Security;
 
 namespace VirtoCommerce.ExperienceApiModule.XOrder.Extensions
 {
     public static class ServiceCollectionExtensions
     {
         public static IServiceCollection AddXOrder(this IServiceCollection services, IGraphQLBuilder graphQlbuilder)
         {
-            services.AddSchemaBuilder<OrderSchema>();
+            services.AddSingleton<ISchemaBuilder>(provider => {
```

Comment:
I would make it transient and remove this factory building. Seems it is not a real singleton object.
But maybe singleton is used for optimization purposes (if creating schema is an expensive operation - though seems it is not)?
@tatarincev How do you think, should we use a singleton for all schemas?